packages/query-devtools/src/Explorer.tsx

```diff
@@ -265,13 +345,48 @@ export default function Explorer(props: ExplorerProps) {
         </Show>
       </Show>
       <Show when={subEntryPages().length === 0}>
-        <div
-          style={{
-            'line-height': '1.125rem',
-          }}
-        >
-          <span class={styles.label}>{props.label}:</span>{' '}
-          <span class={styles.value}>{displayValue(props.value)}</span>
+        <div class={styles.row}>
+          <span class={styles.label}>{props.label}:</span>
+          <Show
+            when={
+              props.editable && (type() === 'string' || type() === 'number')
+            }
+            fallback={
+              <span class={styles.value}>{displayValue(props.value)}</span>
+            }
+          >
+            <input
+              type={type() === 'number' ? 'number' : 'text'}
+              class={cx(styles.value, styles.editableInput)}
+              value={props.value as string} // TODO? can we avoid this?
+              onChange={(changeEvent) => {
+                const oldData = props.activeQuery?.state.data
+
+                const newData = updateNestedDataByPath(
+                  oldData,
+                  currentDataPath,
+                  type() === 'number'
+                    ? changeEvent.target.valueAsNumber
+                    : changeEvent.target.value,
+                )
+
+                props.activeQuery?.setData(newData)
```

Comment:
when calling `setData` like this, we should pass `options` to it as well:

- `manual: true` to avoid messing with `fetchStatus`
- `dataUpdatedAt` should probably be set to the current `dataUpdatedAt` of the query? Otherwise, every write would re-set staleTime counters

Reply:
actually, the `dataUpdatedAt` might not be necessary. I think its fine to treat the data you write in the devtools as `fresh`. Otherwise, refetches might overwrite it too eagerly. But the `manual: true` is something we should do.

If we want to use a different api, it would be:

```
queryClient.setQueryData(props.actieveQuery.queryKey, newData)
```

because that function sets `manual: true` automatically. It's the more "public" api

Reply:
Nice updated the code accordingly